tests/src/Traits/UtilityTrait.php

```diff
@@ -49,6 +49,36 @@ protected function formatMailBodyText(string $mail_body): string {
     return trim($mail_body);
   }
 
+  /**
+   * Retrieves emails sent to the user from the email collection.
+   *
+   * @param string $user
+   *   The user name.
+   *
+   * @return array
+   *   An array of mails as stored in the email collector.
+   *
+   * @throws \Exception
+   *   Thrown if the user is not found or no emails have been sent.
+   */
+  protected function getUserMails(string $user): array {
+    $user = $this->getUserByName($user);
+    if (empty($user)) {
+      throw new \Exception("User {$user->getAccountName()} was not found.");
+    }
+
+    $user_email = $user->getEmail();
+    \Drupal::state()->resetCache();
+    $mails = \Drupal::state()->get('system.test_mail_collector');
+    if (empty($mails)) {
+      throw new \Exception('No mail was sent.');
+    }
```

Comment:
This section is duplicated in a few locations. Let's factor it out in a separate method so it can be reused.

Maybe we can make an `EmailTrait` to collect all reusable code related to inspecting emails?